utils/timeout.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef UTILS_TIMEOUT_H_
+#define UTILS_TIMEOUT_H_
+
+#include <pthread.h>
+#include <signal.h>
+#include <sys/time.h>
+#include "accelerator/common_core.h"
+#include "accelerator/config.h"
+#include "accelerator/errors.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file utils/timeout.h
+ */
+
+typedef struct {
+  const iota_config_t *iconf;
+  const iota_client_service_t *service;
+  ta_generate_address_res_t *res;
+} ta_generate_address_args_t;
+
+pthread_t ta_generate_address_thread_id;
+static const struct itimerval TA_TIMER_TIMEOUT = {.it_interval = {.tv_sec = 0, .tv_usec = 0},
+                                                  .it_value = {.tv_sec = 10, .tv_usec = 0}};
+static const struct itimerval TA_ZERO_TIMEOUT = {.it_interval = {.tv_sec = 0, .tv_usec = 0},
+                                                 .it_value = {.tv_sec = 0, .tv_usec = 0}};
+
+status_t ta_generate_address_helper(const iota_config_t *const iconf, const iota_client_service_t *const service,
```

Comment:
I would expect the availability of a certain function which accepts the given callback function(s) and tracks its valid durations.

Reply:
I've implemented a non thread-safe callback timer which allows a callback to be halted after a period of time. The utility of the function is locked by `cjson_lock`, so there should be only one access to the functions at one time.

Reply:
Since `cjson_lock` was introduced by @jkrvivian, I would defer to her opinions.